gsa/src/gmp/models/__tests__/alert.js

```diff
@@ -0,0 +1,105 @@
+/* Greenbone Security Assistant
+ *
+ * Authors:
+ * Steffen Waterkamp <steffen.waterkamp@greenbone.net>
+ *
+ * Copyright:
+ * Copyright (C) 2018 Greenbone Networks GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import Model from 'gmp/model';
+import Alert from 'gmp/models/alert';
+import {testModel} from 'gmp/models/testing';
+
+testModel(Alert, 'alert');
+
+describe('Alert Model tests', () => {
+
+  test('should parse condition, event, and method', () => {
+    const elem = {
+      condition: {
+        __text: 'text',
+      },
+      event: {
+        __text: 'lorem',
+        data: {
+          __text: 'ipsum',
+          name: 'dolor',
+        },
+      },
+      method: {
+        __text: 'foo',
+        data: {
+          __text: '42',
+          name: 'bar',
+        },
+      },
+    };
+    const alert = new Alert(elem);
+
+    expect(alert.condition).toEqual({
+      data: {},
+      type: 'text',
+    });
+    expect(alert.event).toEqual({
+      data: {
+        dolor: {value: 'ipsum'},
+      },
+      type: 'lorem',
+    });
+    expect(alert.method).toEqual({
+      data: {
+        bar: {value: '42'},
+      },
+      type: 'foo',
+    });
+  });
+
+  test('should return given filter as instance of filter model', () => {
+    const elem = {filter: 'rows=1337'};
+    const alert = new Alert(elem);
+
+    expect(alert.filter).toEqual(new Model('rows=1337', 'filter'));
```

Comment:
Uh maybe it would be better to use a real Filter model here in future. Not sure if that would create a cyclic dependency...